src/deltacode/__init__.py

```diff
@@ -48,14 +50,30 @@ class DeltaCode(object):
     the form of File objects) contained in those scans.
     """
     def __init__(self, new_path, old_path, options):
-        self.new = Scan(new_path)
-        self.old = Scan(old_path)
+        self.codebase1 = None
+        self.codebase2 = None
+        try:
+            self.codebase1 = VirtualCodebase(new_path)
+            self.codebase2 = VirtualCodebase(old_path)
+        except Exception as exception:
+            click.secho(exception.message ,fg = "red")
+        self.new_files_count = 0 #keeps the count of the new file
+        self.old_files_count = 0 #keeps the count of old files
+        self.new_files = [] # a list of [[new file1:Original path],[new file2:Original Path],...]
+        self.old_files = [] # a list of [[old file1:Original path],[old file2:Original Path],...]
+        self.new_files_fingerprint = dict() # map of { {new_file1:fingerprint},{new_file2:fingerprint},...} it will be needed when we need the fingerprints
+        self.old_files_fingerprint = dict() # map of { {old_file1:fingerprint},{old_file2:fingerprint},...}
+        self.new_files_original_path = dict() #this keeps a map of the path of file with respect to original path 
+        self.old_files_original_path = dict()
         self.options = options
         self.deltas = []
         self.errors = []
-        self.stats = Stat(self.new.files_count, self.old.files_count)
-
-        if self.new.path != '' and self.old.path != '':
+       
+        if self.codebase1 != None and self.codebase2 != None:
```

Comment:
Can we just have a check for the inverse of this so we dont write everything under an unneeded indentation?

Reply:
@majurg  yeah  , trying for it , I think it would be a better approach